code/game/objects/items/devices/scanners.dm

```diff
@@ -126,11 +126,20 @@ MASS SPECTROMETER
 	if(M.status_flags & FAKEDEATH)
 		mob_status = "<span class='alert'>Deceased</span>"
 		oxy_loss = max(rand(1, 40), oxy_loss, (300 - (tox_loss + fire_loss + brute_loss))) // Random oxygen loss
-
 	if(ishuman(M))
 		var/mob/living/carbon/human/H = M
-		if(H.heart_attack && H.stat != DEAD)
+		if(H.has_medical_effect(/datum/medical_effect/flatline) && H.stat != DEAD)
 			user << "<span class='danger'>Subject suffering from heart attack: Apply defibrillator immediately!</span>"
+	if(istype(M, /mob/living/carbon/))

```

Comment:
snowflake tbh, add some kind of callback on the mob where it can add medical information to scanner outputs